src/pages/ImportData.js

```diff
@@ -34,32 +37,53 @@ const ChoiceButton = ({ iconName, title, detail, style, ...props }) => {
   return h(Clickable, {
     style: {
       ...style,
-      padding: '1rem',
+      padding: '1rem', marginTop: '1rem',
       display: 'flex', alignItems: 'center',
-      border: `1px solid ${colors.accent()}`, borderRadius: 4
+      border: `1px solid ${colors.accent(1)}`, borderRadius: 4
     },
     hover: { backgroundColor: colors.accent(0.1) },
     ...props
   }, [
-    icon(iconName, { size: 24, style: { flex: 'none', marginRight: '1rem' } }),
+    icon(iconName, { size: 29, style: { flex: 'none', marginRight: '1rem', color: colors.accent(1) } }),
     div({ style: { flex: 1 } }, [
-      div({ style: { fontWeight: 'bold' } }, [title]),
+      div({ style: { fontWeight: 'bold', color: colors.accent(1) } }, [title]),
       div([detail])
     ]),
-    icon('angle-right', { size: 32, style: { flex: 'none', marginLeft: '1rem' } })
+    icon('angle-right', { size: 32, style: { flex: 'none', marginLeft: '1rem', color: colors.accent(1) } })
   ])
 }
 
 const ImportData = () => {
-  const { workspaces, refresh: refreshWorkspaces } = useWorkspaces()
+  const { workspaces, refresh: refreshWorkspaces, loading: loadingWorkspaces } = useWorkspaces()
   const [isImporting, setIsImporting] = useState(false)
-  const { query: { url, format, ad, wid } } = Nav.useRoute()
+  const { query: { url, format, ad, wid, template } } = Nav.useRoute()
   const [mode, setMode] = useState(wid ? 'existing' : undefined)
   const [isCreateOpen, setIsCreateOpen] = useState(false)
+  const [isCloneOpen, setIsCloneOpen] = useState(false)
   const [selectedWorkspaceId, setSelectedWorkspaceId] = useState(wid)
+  const [selectedTemplateWorkspaceKey, setSelectedTemplateWorkspaceKey] = useState()
+  const [allTemplates, setAllTemplates] = useState()
+
+  const noteMessage = 'Note that the import process may take some time after you are redirected into your destination workspace.'
 
   const selectedWorkspace = _.find({ workspace: { workspaceId: selectedWorkspaceId } }, workspaces)
 
+  const filteredTemplates =
+    _.flow(
+      _.flatMap(id => (allTemplates && allTemplates[id]) || []),
+      _.filter(({ name, namespace }) => _.some({ workspace: { namespace, name } }, workspaces))
+    )(_.isArray(template) ? template : Array.of(template))
+
+  Utils.useOnMount(() => {
+    const loadTemplateWorkspaces = _.flow(
+      Utils.withBusyState(setIsImporting),
+      withErrorReporting('Import Error')
```

Comment:
This error message could be more descriptive, something like 'Error loading templates'